From 3f3e6bec92c29575e97e80ef295e5dd71b22a785 Mon Sep 17 00:00:00 2001
From: Jose Lamego <jose.a.lamego@intel.com>
Date: Fri, 27 Apr 2018 10:00:54 -0500
Subject: [PATCH] use iso-info command

iso files cannot be handled due to isoinfo command
not found.

This change uses iso-info command instead.

Signed-off-by: Jose Lamego <jose.a.lamego@intel.com>

---
Rebased patch for update to 2.1.0

Signed-off-by: Simental Magana, Marcos <marcos.simental.magana@intel.com>
---
 tests/clitest.py       |  2 +-
 virtinst/urlfetcher.py | 14 +++++++-------
 2 files changed, 8 insertions(+), 8 deletions(-)

diff --git a/tests/clitest.py b/tests/clitest.py
index 504b487..38b0862 100644
--- a/tests/clitest.py
+++ b/tests/clitest.py
@@ -29,7 +29,7 @@ os.environ["DISPLAY"] = ":3.4"
 TMP_IMAGE_DIR = "/tmp/__virtinst_cli_"
 XMLDIR = "tests/cli-test-xml"
 OLD_OSINFO = utils.has_old_osinfo()
-HAS_ISOINFO = find_executable("isoinfo")
+HAS_ISOINFO = find_executable("iso-info")
 
 # Images that will be created by virt-install/virt-clone, and removed before
 # each run
diff --git a/virtinst/urlfetcher.py b/virtinst/urlfetcher.py
index 94a8aa2..f653240 100644
--- a/virtinst/urlfetcher.py
+++ b/virtinst/urlfetcher.py
@@ -298,26 +298,26 @@ class _ISOURLFetcher(_URLFetcher):
 
     def _grabber(self, url):
         """
-        Use isoinfo to grab the file
+        Use iso-info to grab the file
         """
         if not self._hasFile(url):
-            raise RuntimeError("isoinfo didn't find file=%s" % url)
+            raise RuntimeError("iso-info didn't find file=%s" % url)
 
-        cmd = ["isoinfo", "-J", "-i", self.location, "-x", url]
+        cmd = ["iso-info", "-i", self.location, "-x", url]
 
-        logging.debug("Running isoinfo: %s", cmd)
+        logging.debug("Running iso-info: %s", cmd)
         output = subprocess.check_output(cmd)
 
         return io.BytesIO(output), len(output)
 
     def _hasFile(self, url):
         """
-        Use isoinfo to list and search for the file
+        Use iso-info to list and search for the file
         """
         if not self._cache_file_list:
-            cmd = ["isoinfo", "-J", "-i", self.location, "-f"]
+            cmd = ["iso-info", "-i", self.location, "-f"]
 
-            logging.debug("Running isoinfo: %s", cmd)
+            logging.debug("Running iso-info: %s", cmd)
             output = subprocess.check_output(cmd, stderr=subprocess.DEVNULL)
 
             self._cache_file_list = output.splitlines(False)
-- 
2.21.0

